### ISSUE TITLE
Remove implementation of `/v1/oauth/tokens` endpoint

### ISSUE THREAD

**Issue by jbonofre** [polaris-catalog/polaris#12]
The oauth/tokens endpoint is already deprecated for removal due to security concerns. The path forward is described in [this doc](https://docs.google.com/document/d/1Xi5MRk8WdBWFC3N_eSmVcrLhk3yu5nJ9x_wC0ec6kVQ/).

**Comment by flyrain** [comment]
We don't recommend anyone to use it in a prod env, but it's a nice feature to allow people to try Polaris out, otherwise they have to hook a third-party token service before doing anything else. What we can do here is to remove it while providing an private endpoint with the same functionality for test purpose. In that case, the real prod user can hook their token service, while we still use the private endpoint for testing.

**Comment by dimas-b** [comment]
It is pretty easy to setup Keycloak, for example, as an IdP for "getting started" environments. 

Nessie has an example here: https://github.com/projectnessie/nessie/tree/main/docker/authn-keycloak

**Comment by dimas-b** [comment]
> providing an private endpoint with the same functionality for test purpose

I'm not sure it is as easy as it sounds. A proper OAuth2 implementation is not a small effort. On the other hand, a test impl. that has gaps wrt to the OAuth2 RFC can lead to bugs in production code because it would be tested against a non-compliant OAuth2 server.

I believe it is preferable to reuse existing OSS implementations that provide OAuth2.

**Comment by dimas-b** [comment]
A tangential point: if Quarkus were used as the server platform, it provides easy ways to test authN/Z in CI without requiring a full OAuth2/OIDC implementation.

**Comment by flyrain** [comment]
Agreed, it's great to reuse an existing OAuth2 implementation. We will need to see how integration works to choose between them.

**Comment by rmannibucau** [comment]
Hi,

would a security-less solution be enabled?

there are cases the security is not needed - secured by network whatever it means (k8s policies, aws policies etc), and having to add a server is overkill and complex for literally no gain

also keycloak must not be seen as an easy solution since its clustering requires some advanced knowledge and infra setup to work cleanly

dex is easier to setup but still require an effort for cases the catalog is used in an isolated environment so I hope a trivial plug and play solution remains - once again a noop impl is fine, or just keeping the endpoint with an explicit toggle

hope it makes sense

**Comment by jbonofre** [comment]
@rmannibucau good points. The intend here is regarding the IRC spec. We can totally have a Polaris embedded solution to behave the same.
